Modules/_csv.c

```diff
@@ -1348,16 +1346,20 @@ Writer_clear(WriterObj *self)
     Py_CLEAR(self->dialect);
     Py_CLEAR(self->write);
     Py_CLEAR(self->error_obj);
+    if (self->rec != NULL) {
```

Comment:
This is incorrect, `tp_clear` should not free memory, just break inter-object references.

Reply:
Only `tp_dealloc` should be freeing memory.

Reply:
Got it. I'll move them to dealloc. Thanks!

Reply:
Fixed in fe74fc2dd9ca1a59e3f12a0bb5adaf0c36834b85